features/poll/impl/src/main/kotlin/io/element/android/features/poll/impl/create/CreatePollPresenter.kt

```diff
@@ -0,0 +1,168 @@
+                            pollKind = pollKind,
+                        )
+                        // analyticsService.capture(PollCreate()) // TODO: Send PollCreate analytics.
+                        navigateUp()
+                    } else {
+                        Timber.e("Cannot create poll")
+                    }
+                }
+                is CreatePollEvents.AddAnswer -> {
+                    answers = answers + ""
+                }
+                is CreatePollEvents.RemoveAnswer -> {
+                    answers = answers.filterIndexed { index, _ -> index != event.index }
+                }
+                is CreatePollEvents.SetAnswer -> {
+                    val text = if (event.text.length > MAX_ANSWER_LENGTH) {
+                        event.text.substring(0, MAX_ANSWER_LENGTH)
+                    } else {
+                        event.text
+                    }
+                    answers = answers.toMutableList().apply {
+                        this[event.index] = text
+                    }
+                }
+                is CreatePollEvents.SetPollKind -> {
+                    pollKind = event.pollKind
+                }
+                is CreatePollEvents.SetQuestion -> {
+                    question = event.question
+                }
+                is CreatePollEvents.NavBack -> {
+                    navigateUp()
+                }
+                CreatePollEvents.ConfirmNavBack -> {
+                    val shouldConfirm = question.isNotBlank() || answers.any { it.isNotBlank() }
+                    if (shouldConfirm) {
+                        showConfirmation = true
+                    } else {
+                        navigateUp()
+                    }
+                }
+                is CreatePollEvents.HideConfirmation -> showConfirmation = false
+            }
+        }
+
+        return CreatePollState(
+            canCreate = canCreate,
+            canAddAnswer = canAddAnswer,
+            question = question,
+            answers = immutableAnswers,
+            pollKind = pollKind,
+            showConfirmation = showConfirmation,
+            eventSink = ::handleEvents,
+        )
+    }
+}
+
+private fun canCreate(
+    question: String,
+    answers: List<String>
+) = question.isNotBlank() && answers.size >= MIN_ANSWERS && answers.all { it.isNotBlank() }
+
+private fun canAddAnswer(answers: List<String>) = answers.size < MAX_ANSWERS
+
+private fun List<String>.toAnswers(): ImmutableList<Answer> {
+    return map { answer ->
+        Answer(
+            text = answer,
+            canDelete = this.size > MIN_ANSWERS,
+        )
+    }.toImmutableList()
+}
+
+private val pollKindSaver: Saver<MutableState<PollKind>, Boolean> = Saver(
+    save = {
+        when (it.value) {
+            PollKind.Disclosed -> false
+            PollKind.Undisclosed -> true
+        }
+    },
+    restore = {
+        mutableStateOf(
+            when {
```

Comment:
There is a problem here, you are not using `it`:

```suggestion
            when(it) {
```

Reply:
Oh damn, thanks, this is fatigue, I was wondering all day why intellij was underlying that `when` and couldn't fathom.